crates/pevm/src/lib.rs

```diff
@@ -4,14 +4,49 @@
 
 // TODO: Better types & API for third-party integration
 
-use std::hash::{BuildHasher, BuildHasherDefault, Hash, Hasher};
+use std::{
+    future::Future,
+    hash::{BuildHasher, BuildHasherDefault, Hash, Hasher},
+    sync::OnceLock,
+    thread,
+};
+use tokio::runtime::{Handle, Runtime};
 
 use alloy_primitives::{Address, B256, U256};
 use bitflags::bitflags;
 use hashbrown::HashMap;
 use rustc_hash::FxBuildHasher;
 use smallvec::SmallVec;
 
+static GLOBAL_RUNTIME: OnceLock<Runtime> = OnceLock::new();
+
+#[inline(always)]
+/// Awaits on a future using the current Tokio runtime if it exists,
+/// or creates a new one (at most once) if it doesn't.
+pub fn block_on<F: Future + Send>(future: F) -> F::Output
+where
+    F::Output: Send,
+{
+    // Get a handle to the current Tokio runtime
+    if Handle::try_current().is_ok() {
+        thread::scope(|s| {
```

Comment:
This is a cool trick to avoid creating a new runtime that blocks in the main thread of another runtime! But I guess the more "standard" solution is to use `tokio::task::block_in_place`. It helps remove the `Send` requirement as well.

Reply:
@hai-rise it was a hack to prevent the panics of `block_in_place` due to `tokio::test`, but I'll remove it.

Reply:
Yeah, but it would still deadlock in a current-thread runtime if it is used to drive another asynchronous task (like fetching the full block before executing it). If anything `block_in_place`'s panic message suggests library users switch to a multi-thread runtime!

Let's settle on not supporting current-thread runtime and either:
- Build `RpcStorage`'s own multi-thread runtime if there is not an active one; or
- Expect the active runtime to be multi-thread.